c/new.go

```diff
@@ -0,0 +1,126 @@
+package c
+
+import (
+	"context"
+	"time"
+
+	"github.com/capatazlib/go-capataz/internal/c"
+)
+
+// NotifyStartFn is a function given to supervisor children to notify the
+// supervisor that the child has started.
+//
+// ### Notify child's start failure
+//
+// In case the child cannot get started it should call this function with an
+// error value different than nil.
+//
+type NotifyStartFn = c.NotifyStartFn
+
+// ChildSpec represents a Child specification; it serves as a template for the
+// construction of a goroutine. The ChildSpec record is used in conjunction with
+// the supervisor's SupervisorSpec.
+//
+// # A note about ChildTag
+//
+// An approach that we considered was to define a type heriarchy for
+// SupervisorChildSpec and WorkerChildSpec to deal with differences between
+// Workers and Supervisors rather than having a value you can use in a switch
+// statement. In reality, the differences between the two are minimal (only
+// behavior change happens when sending notifications to the events system). If
+// this changes, we may consider a design where we have a ChildSpec interface
+// and we have different implementations.
+type ChildSpec = c.ChildSpec
+
+// New creates a `ChildSpec` that represents a worker goroutine. It requires two
+// arguments: a `name` that is used for runtime tracing and a `startFn` function.
+//
+// ### The `name` argument
+//
+// The `name` argument must not be empty nor contain forward slash characters
+// (e.g. `/`), otherwise, the system will panic. This method is preferred as
+// opposed to return an error given it is considered a bad implementation
+// (ideally a compilation error).
+//
+// ### The `startFn` argument
+//
+// The `startFn` function attribute of a `ChildSpec` is going to be used to
+// spawn a new supervised goroutine; in this function is where your business
+// logic should be located.
```

Comment:
I would reorder this a little. Consider this text:
```
// The `startFn` is the function where your business logic should be located.
// This attribute of a `ChildSpec` is going to be used to spawn a new supervised
// goroutine
```
They key bit is the business logic